### PR TITLE
feat: Google Sheets UQI migration

### PR DESCRIPTION
**Pull request by nidhi-nair**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.

## Description

> Please include a summary of the changes and which issue has been fixed. Please also include relevant motivation
> and context. List any dependencies that are required for this change.

Fixes # (issue)

> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: task/gs-uqi-2 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.52 **(0.06)** | 37.95 **(0.06)** | 36.19 **(0.07)** | 56.75 **(0.07)**
 :green_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0)** | 72.55 **(1.96)** | 100 **(0)** | 93.33 **(0)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :sparkles: :new: | **app/client/src/workers/ReplayDSL.ts** | **91.8** | **76.47** | **87.5** | **91.38**
 :sparkles: :new: | **app/client/src/workers/replayUtils.ts** | **91.3** | **75** | **100** | **95.12**</details>

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ClientDataDisplayType.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 package com.appsmith.external.models;
 
 public enum ClientDataDisplayType {
```
Comment (sumitsum) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ClientDataDisplayType.java:3]: Can you please add a comment here explaining what it does ?

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/Condition.java [hunk 2]
```diff
@@ -81,7 +81,9 @@ public static Condition addValueDataType(Condition condition) {
      */
     public static Boolean isValid(Condition condition) {
 
-        if (StringUtils.isEmpty(condition.getPath()) || (condition.getOperator() == null)) {
+        if (condition == null ||
```
Comment (sumitsum) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/Condition.java:84]: Can you please update the comment to indicate why we are looking at value as well ?

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/TriggerRequestDTO.java [hunk 3]
```diff
@@ -5,21 +5,21 @@
...
 import lombok.Setter;
 
-import java.util.List;
+import java.util.Map;
 
 @NoArgsConstructor
 @AllArgsConstructor
 @Getter
 @Setter
 public class TriggerRequestDTO {
```
Comment (sumitsum) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/TriggerRequestDTO.java:14]: Can you please add comment explaining what it does ?

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/ce/FilterDataServiceCE.java [hunk 4]
```diff
@@ -195,7 +151,7 @@ private List<Map<String, Object>> executeFilterQueryNew(String tableName, Map<St
          */
         List<PreparedStatementValueDTO> values = new ArrayList<>();
 
-        if (condition != null) {
+        if (condition != null && Condition.isValid(condition)) {
```
Comment (sumitsum) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/ce/FilterDataServiceCE.java:154]: Minor: Is the `null` check redundant here ?

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java [hunk 5]
```diff
@@ -476,7 +478,7 @@ private Mono<ActionExecutionResult> executeCommon(AmazonS3 connection,
 
                         Map<String, Object> formData = actionConfiguration.getFormData();
 
-                        String command = (String) getValueSafelyFromFormData(formData, COMMAND);
+                        String command = PluginUtils.getDataValueSafelyFromFormData(formData, COMMAND, STRING_TYPE);
```
Comment (sumitsum) [inline app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:481]: Minor: would it be possible to write it without appending `PluginUtils.` ?

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/TriggerMethod.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.external.config;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.web.reactive.function.client.WebClient;
+
+public interface TriggerMethod {
```
Comment (sumitsum) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/TriggerMethod.java:6]: Can you please write a comment explaining what this class does ?

File: app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/TriggerMethod.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+    boolean validateTriggerMethodRequest(MethodConfig methodConfig);
```
Comment (sumitsum) [inline app/server/appsmith-plugins/googleSheetsPlugin/src/main/java/com/external/config/TriggerMethod.java:8]: Can you please write comments explaining each of the method ?

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java (hunk 5 above)
Comment (nidhi-nair) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:481]: Hey @sumitsum ! Could you please help me understand what the rationale for doing this is?

Comment (sumitsum) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java]: @nidhi-nair just to improve the code reading / writing experience. What I meant was that if we could import like `import PluginUtils.getData...` then we could use the function directly.

Comment (nidhi-nair) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java]: Gotcha, thanks for clarifying. I've updated the references across. :)

**Pull request closed without merging**